src/libraries/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/common/DirectoryControl.cs

```diff
@@ -132,102 +145,174 @@ public virtual byte[] GetValue()
 
         internal static void TransformControls(DirectoryControl[] controls)
         {
-            for (int i = 0; i < controls.Length; i++)
+            Span<byte> attributeNameScratchSpace = stackalloc byte[AttributeNameStackAllocationThreshold];
+
+            try
             {
-                Debug.Assert(controls[i] != null);
-                byte[] value = controls[i].GetValue();
-                if (controls[i].Type == "1.2.840.113556.1.4.319")
+                for (int i = 0; i < controls.Length; i++)
                 {
-                    // The control is a PageControl.
-                    object[] result = BerConverter.Decode("{iO}", value);
-                    Debug.Assert((result != null) && (result.Length == 2));
+                    Debug.Assert(controls[i] != null);
+                    byte[] value = controls[i]._directoryControlValue ?? Array.Empty<byte>();
```

Comment:
When can `controls[i]._directoryControlValue` be null?

There is no test coverage for this branch.

Reply:
I don't think we need a test if you removed the check - I was only verifying test coverage.

Reply:
Excellent - thanks for the review!